modules/nw-configuring-ingress-cluster-traffic-aws-networking-load-balancer.adoc

```diff
@@ -0,0 +1,9 @@
+// Module included in the following assemblies:
+//
+// * networking/configuring_ingress_cluster_traffic/configuring-ingress-cluster-traffic-aws.adoc
+
+:_content-type: CONCEPT
+[id="nw-configuring-ingress-cluster-traffic-aws-network-load-balancer_{context}"]
+= Configuring ingress cluster traffic on AWS using a Network Load Balancer
+
+{product-title} provides methods for communicating from outside the cluster with services running in the cluster. This method uses a Network Load Balancer (NLB), which forwards the client's IP address to the node. You can configure an NLB on a new or existing AWS cluster.
```

Comment:
The phrase "which forwards the client's IP address to the node" seems out of place.  I believe that it is referring to [Source Address Preservation](https://aws.amazon.com/blogs/aws/new-network-load-balancer-effortless-scaling-to-millions-of-requests-per-second/).  That is indeed a feature of NLBs, but the same end result is possible with CLBs using [AWS's support for Proxy Protocol](https://aws.amazon.com/blogs/aws/elastic-load-balancing-adds-support-for-proxy-protocol/), and source address preservation is only one of several features of NLBs.  

Is there supposed to be more content in this module to show how to create a new IngressController that uses an NLB, or is the module intended to lead into the "Replacing Ingress Controller Classic Load Balancer with Network Load Balancer" module?

Reply:
This is one of those things that `git` has marked as a file change, but [the content itself has not been edited](https://docs.openshift.com/container-platform/4.10/networking/configuring_ingress_cluster_traffic/configuring-ingress-cluster-traffic-aws-network-load-balancer.html), only modularized (it was a top-level assembly prior to this restructure). 

Would it be better if this text were omitted entirely? Or perhaps moved to the beginning of the assembly, just under the H1 heading "Configuring ingress cluster traffic on AWS"? That location might be a more natural fit, since both NLBs and CLBs can be mentioned there together.

> Is there supposed to be more content in this module to show how to create a new IngressController that uses an NLB, or is the module intended to lead into the "Replacing Ingress Controller Classic Load Balancer with Network Load Balancer" module?

Yeah it's the latter: the module is meant as a lead-in to the following procedures: 
- "Replacing Ingress Controller Classic Load Balancer with Network Load Balancer"
- "Configuring an Ingress Controller Network Load Balancer on an existing AWS cluster"

Reply:
> This is one of those things that `git` has marked as a file change, but [the content itself has not been edited](https://docs.openshift.com/container-platform/4.10/networking/configuring_ingress_cluster_traffic/configuring-ingress-cluster-traffic-aws-network-load-balancer.html), only modularized (it was a top-level assembly prior to this restructure).

Sorry, my mistake.  If it doesn't make sense to change the text as part of this PR, please disregard my comment.  

> Would it be better if this text were omitted entirely? Or perhaps moved to the beginning of the assembly, just under the H1 heading "Configuring ingress cluster traffic on AWS"? That location might be a more natural fit, since both NLBs and CLBs can be mentioned there together.

It could be omitted, or your idea to move it under the heading makes a lot of sense.  We preserve the source address with both CLBs and NLBs, but the mechanisms are different.  CLBs use [PROXY protocol](https://github.com/haproxy/haproxy/blob/master/doc/proxy-protocol.txt), which wraps TCP connections by adding a header with the source address, which OpenShift can read when it receives the connection from the CLB, whereas NLBs provide transparent source preservation, in which the TCP connection has the original client's source address when OpenShift receives the connection from the NLB.  In either case, the end result is that we put the original client's address in HTTP headers and access logs.  

> > Is there supposed to be more content in this module to show how to create a new IngressController that uses an NLB, or is the module intended to lead into the "Replacing Ingress Controller Classic Load Balancer with Network Load Balancer" module?
> 
> Yeah it's the latter: the module is meant as a lead-in to the following procedures:
> 
>     * "Replacing Ingress Controller Classic Load Balancer with Network Load Balancer"
> 
>     * "Configuring an Ingress Controller Network Load Balancer on an existing AWS cluster"

All right, thanks for confirming!  It just looks a little funny when looking at the module in isolation, but it makes sense with the context in the assembly.

Reply:
Thanks @Miciah ! I've omitted the text in this file, but added some additional text and a link in `networking/configuring_ingress_cluster_traffic/configuring-ingress-cluster-traffic-aws.adoc`